catalog/core/catalog-core-standardframework/src/test/java/ddf/catalog/cache/CacheKeyTest.java

```diff
@@ -38,166 +38,165 @@
+    public void testKeyGenerationFromMetacardAndResourceRequest() {
         CacheKey cacheKey = new CacheKey(getMetacardStub("sampleId"), getResourceRequestStub());
 
-        // when
         String key = cacheKey.generateKey();
 
-        // then
         assertNotNull("Key must not be null.", key);
         assertThat("Key must not be empty.", key, not(equalToIgnoringWhiteSpace("")));
+    }
+
+    @Test
+    public void testKeyGenerationFromMetacardOnly() {
+        CacheKey cacheKey = new CacheKey(getMetacardStub("sampleId"));
+
+        String key = cacheKey.generateKey();
 
+        assertNotNull("Key must not be null.", key);
+        assertThat("Key must not be empty.", key, not(equalToIgnoringWhiteSpace("")));
     }
 
-    @Test()
+    @Test
     public void testKeyUniquenessMetacardId() {
-
-        // given
         CacheKey cacheKey1 = new CacheKey(getMetacardStub("sampleId"), getResourceRequestStub());
         CacheKey cacheKey2 = new CacheKey(getMetacardStub("sampledI"), getResourceRequestStub());
 
-        // when
         String key1 = cacheKey1.generateKey();
         String key2 = cacheKey2.generateKey();
 
-        // then
         assertThat("Keys must be different.", key1, not(equalTo(key2)));
-
     }
 
-    @Test()
+    @Test
     public void testKeyUniquenessFromSources() {
-
-        // given
         CacheKey cacheKey1 = new CacheKey(getMetacardStub("sampleId", "source1"),
                 getResourceRequestStub());
         CacheKey cacheKey2 = new CacheKey(getMetacardStub("sampleId", "source2"),
                 getResourceRequestStub());
 
-        // when
         String key1 = cacheKey1.generateKey();
         String key2 = cacheKey2.generateKey();
 
-        // then
         assertThat("Keys must be different.", key1, not(equalTo(key2)));
+    }
+
+    @Test
+    public void testKeyUniquenessFromSourcesWithoutResourceRequest() {
+        CacheKey cacheKey1 = new CacheKey(getMetacardStub("sampleId", "source1"));
+        CacheKey cacheKey2 = new CacheKey(getMetacardStub("sampleId", "source2"));
+
+        String key1 = cacheKey1.generateKey();
+        String key2 = cacheKey2.generateKey();
 
+        assertThat("Keys must be different.", key1, not(equalTo(key2)));
     }
 
-    @Test()
+    @Test
     public void testKeyUniquenessFromSourcesAndIds() {
-
-        // given
         CacheKey cacheKey1 = new CacheKey(getMetacardStub("sampleId1", "source1"),
                 getResourceRequestStub());
         CacheKey cacheKey2 = new CacheKey(getMetacardStub("sampleId2", "source2"),
                 getResourceRequestStub());
 
-        // when
         String key1 = cacheKey1.generateKey();
         String key2 = cacheKey2.generateKey();
 
-        // then
         assertThat("Keys must be different.", key1, not(equalTo(key2)));
+    }
+
+    @Test
+    public void testKeyUniquenessFromSourcesAndIdsWithoutResourceRequest() {
+        CacheKey cacheKey1 = new CacheKey(getMetacardStub("sampleId1", "source1"));
```

Comment:
Might be worth adding two more keys and assertions:
```java
CacheKey cacheKey1b = new CacheKey(getMetacardStub("sampleId1", "source2"));
CacheKey cacheKey1c = new CacheKey(getMetacardStub("sampleId2", "source1"));
String key1b = cacheKey1b.generateKey();
String key1c = cacheKey1c.generateKey();

assertThat("Keys must be different.", key1, not(equalTo(key1b)));
assertThat("Keys must be different.", key1, not(equalTo(key1c)));
```

That would validate that you're not doing your equality test on only one of the two provided metacard fields.